lib/active_record/connection_adapters/sqlserver_column.rb

```diff
@@ -1,9 +1,12 @@
 module ActiveRecord
   module ConnectionAdapters
     class SQLServerColumn < Column
+      attr_reader :table_name
 
       def initialize(name, default, sql_type_metadata = nil, null = true, table_name = nil, default_function = nil, collation = nil, comment = nil, sqlserver_options = {})
         @sqlserver_options = sqlserver_options || {}
+        @table_name = table_name
```

Comment:
We could get `table_name` from `@sqlserver_options` instead to avoid distancing even more from the original `ActiveRecord` constructor.

You would just need to just need to update `ActiveRecord::ConnectionAdapters::SQLServer::SchemaStatements#columns` to also slice `table_name` and create a  `table_name` method here delegating to `@sqlserver_options` as we do for `is_primary?` for example.

Reply:
Done